`Advanced Table` Expand Collapse All Documentation (#2748)
hashicorp/design-system: website/docs/components/table/advanced-table/partials/guidelines/guidelines.md

!!!

### Expand/Collapse All Button
The Expand/Collapse All button allows users to expand or collapse all rows, including nested rows. It provides quick access to more content but may impact readability when content is long or detailed.

> ### Interactions

Reviewer: ```suggestion
#### Interactions
```

> #### Default state

Reviewer: ```suggestion
##### Default state
```

The Advanced Table supports any combination of expanded or collapsed rows on load. The button reflects the initial state.

> ![Advanced Table with default height parent rows and short density nested rows.](/assets/components/table/advanced-table/expandable-rows-expand-all.png)

Reviewer 2: ```suggestion
![Expand all button with an expand icon.](/assets/components/table/advanced-table/expandable-rows-expand-all.png)
```

> <Doc::ImageCaption @text="Expand All if any rows are collapsed."/>

Reviewer 3: Nit: I would probably just say "Expand all rows" for simplicity. (vs. "Expand All if any rows are collapsed")

Reviewer 3: I think I have better suggestions for these... Working on it...

> <Doc::ImageCaption @text="Expand All if any rows are collapsed."/>

Reviewer 3: ```suggestion
<Doc::ImageCaption @text='“Expand All” button displays if any rows are currently collapsed'/>
```

Reviewer 3: If desired, add "The" to the beginning of the above phrase to make a complete sentence and then add back a period at the end.

Reviewer 3: You could also remove the quotes around the button names, if you wish, but I thought they helped provide clarity.

![Expand all button with a collapse icon.](/assets/components/table/advanced-table/expandable-rows-collapse-all.png)

> <Doc::ImageCaption @text="Collapse All if all rows are expanded."/>

Reviewer 3: ```suggestion
<Doc::ImageCaption @text="“Collapse All” button displays if any rows are currently uncollapsed."/>
```

> ![Advanced Table with default height parent rows and short density nested rows.](/assets/components/table/advanced-table/expandable-rows-collapse-all.png)

Reviewer 2: ```suggestion
![Expand all button with a collapse icon.](/assets/components/table/advanced-table/expandable-rows-collapse-all.png)
```

> ### Collapsed state

Reviewer: ```suggestion
##### Collapsed state
```

> <Doc::ImageCaption @text="“Collapse All” button displays if any rows are currently uncollapsed." />

Reviewer 2: "uncollapsed" is causing a spelling error, also collapse all only shows if every row is expanded.

```suggestion
<Doc::ImageCaption @text="“Collapse All” button displays if all rows are expanded." />
```

![](/assets/components/table/advanced-table/expandable-rows-expand-state.png)

> <Doc::ImageCaption @text="Clicking Expand All expands all rows, including nested rows."/>

Reviewer 3: ```suggestion
<Doc::ImageCaption @text="Clicking “Expand All” expands all rows, including nested rows."/>
```

Wrapping the text you want users to click in quotes would help improve clarity in my opinion.

> ### Expanded state

Reviewer: ```suggestion
##### Expanded state
```

> ![Advanced Table with default height parent rows and short density nested rows.](/assets/components/table/advanced-table/expandable-rows-expand-state.png)

Reviewer 2: ```suggestion
![](/assets/components/table/advanced-table/expandable-rows-expand-state.png)
```

![](/assets/components/table/advanced-table/expandable-rows-collapse-state.png)
<Doc::ImageCaption @text="Once all rows are expanded, the Collapse All button is displayed."/>

> ### Mixed state 

Reviewer: ```suggestion
##### Mixed state 
```

> <Doc::ImageCaption @text="Once all rows are expanded, the Collapse All button is displayed."/>

Reviewer 2: ```suggestion
<Doc::ImageCaption @text="Once all rows are expanded, the “Collapse All” button is displayed."/>
```

> The Advanced Table supports any combination of expanded or collapsed rows on load. The button reflects the initial state.
> 
> ![Expand all button with an expand icon.](/assets/components/table/advanced-table/expandable-rows-expand-all.png)
> <Doc::ImageCaption @text="Expand All if any rows are collapsed."/>
> 
> ![Expand all button with a collapse icon.](/assets/components/table/advanced-table/expandable-rows-collapse-all.png)
> <Doc::ImageCaption @text="Collapse All if all rows are expanded."/>
> 
> ### Collapsed state
> 
> ![](/assets/components/table/advanced-table/expandable-rows-expand-state.png)
> <Doc::ImageCaption @text="Clicking Expand All expands all rows, including nested rows."/>
> 
> ### Expanded state
> 
> ![](/assets/components/table/advanced-table/expandable-rows-collapse-state.png)
> <Doc::ImageCaption @text="Clicking Expand All expands all rows, including nested rows."/>
> 
> ### Mixed state 
> 
> ![](/assets/components/table/advanced-table/expandable-rows-mixed-state.png)

Author: ```suggestion
The Advanced Table supports any combination of expanded or collapsed rows on load. The button reflects the initial state.

![Expand all button with an expand icon.](/assets/components/table/advanced-table/expandable-rows-expand-all.png)
<Doc::ImageCaption @text="Expand All if any rows are collapsed."/>

![Expand all button with a collapse icon.](/assets/components/table/advanced-table/expandable-rows-collapse-all.png)
<Doc::ImageCaption @text="Collapse All if all rows are expanded."/>

### Collapsed state

![](/assets/components/table/advanced-table/expandable-rows-expand-state.png)
<Doc::ImageCaption @text="Clicking Expand All expands all rows, including nested rows."/>

### Expanded state

![](/assets/components/table/advanced-table/expandable-rows-collapse-state.png)
<Doc::ImageCaption @text="Once all rows are expanded, the Collapse All button is displayed."/>

### Mixed state 

![](/assets/components/table/advanced-table/expandable-rows-mixed-state.png)
```

> ![Advanced Table with default height parent rows and short density nested rows.](/assets/components/table/advanced-table/expandable-rows-collapse-state.png)

Reviewer 2: ```suggestion
![](/assets/components/table/advanced-table/expandable-rows-collapse-state.png)
```

> <Doc::ImageCaption @text="Clicking Expand All expands all rows, including nested rows."/>

Author: ```suggestion
<Doc::ImageCaption @text="If some rows are expanded and others are collapsed, the Expand All button will persist until all rows are expanded."/>
```

> <Doc::ImageCaption @text="If some rows are expanded and others are collapsed, the Expand All button will persist until all rows are expanded."/>

Reviewer 2: ```suggestion
<Doc::ImageCaption @text="If some rows are expanded and others are collapsed, the “Expand All” button will persist until all rows are expanded."/>
```

> ![Advanced Table with default height parent rows and short density nested rows.](/assets/components/table/advanced-table/expandable-rows-mixed-state.png)

Reviewer 2: ```suggestion
![](/assets/components/table/advanced-table/expandable-rows-mixed-state.png)
```

> <Doc::ImageCaption @text="Clicking Expand All expands all rows, including nested rows."/>

Reviewer 2: Suggestion: it seems a bit repetitive to have the same caption for all 3 images. Could they be removed if there is not anything else to add?
